internal/service/advancedcluster/resource_advanced_cluster_migration_test.go

```diff
@@ -66,3 +68,106 @@ func TestAccMigrationAdvancedCluster_multiCloud(t *testing.T) {
 		},
 	})
 }
+
+func TestAccMigrationAdvancedClusterAdvancedCluster_partialAdvancedConf(t *testing.T) {
+	var (
+		orgID       = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName = acc.RandomProjectName()
+		rName       = acc.RandomClusterName()
+		processArgs = `advanced_configuration  {
+			fail_index_key_too_long              = false
+			javascript_enabled                   = true
+			minimum_enabled_tls_protocol         = "TLS1_1"
+			no_table_scan                        = false
+		  }`
+		biConnector = `bi_connector_config {
+			enabled = true
+		  }`
+		processArgsUpdated = `advanced_configuration  {
+			fail_index_key_too_long              = false
+			javascript_enabled                   = true
+			minimum_enabled_tls_protocol         = "TLS1_1"
+			no_table_scan                        = false
+			default_read_concern                 = "available"
+			sample_size_bi_connector			 = 110
+            sample_refresh_interval_bi_connector = 310
+		  }`
+		biConnectorUpdated = `bi_connector_config {
+			enabled = false
+			read_preference = "secondary"
+		  }`
+		config        = configPartialAdvancedConfig(orgID, projectName, rName, processArgs, biConnector)
+		configUpdated = configPartialAdvancedConfig(orgID, projectName, rName, processArgsUpdated, biConnectorUpdated)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acc.PreCheckBasic(t) },
+		CheckDestroy: acc.CheckDestroyCluster,
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: mig.ExternalProviders(),
+				Config:            config,
+				Check: resource.ComposeTestCheckFunc(
+					checkExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.0.fail_index_key_too_long", "false"),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.0.javascript_enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.0.minimum_enabled_tls_protocol", "TLS1_1"),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.0.no_table_scan", "false"),
+					resource.TestCheckResourceAttr(resourceName, "bi_connector_config.0.enabled", "true"),
+				),
+			},
+			mig.TestStepCheckEmptyPlan(config),
+			{
```

Comment:
not against but it looks like adding steps after mig one is like mixing mig and acc test together

Reply:
The intention is here is to not only ensure that plans are empty but also that updates and everything work as expected.